Cabal/src/Distribution/Utils/Structured.hs

```diff
@@ -292,6 +294,24 @@ structuredDecodeOrFailIO bs =
     handler (ErrorCall str) = return $ Left str
 #endif
 
+structuredDecodeTriple
+  :: forall a b c. (Structured (a,b,c), Binary.Binary a, Binary.Binary b, Binary.Binary c)
+  => LBS.ByteString -> Either String (a, b, Either String c)
+structuredDecodeTriple lbs =
```

Comment:
Nice! Could you add a comment somewhere explaining why this is necessary (essentially the pr description)?

Reply:
done.